project.clj

```diff
@@ -76,7 +76,7 @@
 
   :plugins [[lein-release "1.0.5" :exclusions [org.clojure/clojure]]]
 
-  :uberjar-name "puppet-server-release.jar"
+  :uberjar-name "puppetserver-release.jar"
```

Comment:
Yeah, fair question.  I tried this out on CentOS 7:

1) Installed puppetserver 2.3.1.
2) Started the puppetserver service and did a successful local agent run against the master.
3) Upgraded to the custom puppetserver build - 2.4.0.master.SNAPSHOT.2016.04.20T0815 (via 'yum upgrade puppetserver').
4) Observed that the old puppetserver service automatically stopped and restarted with the new uberjar automatically, near the end of running the 'yum upgrade' command.
5) Did another successful local agent run against the master.
6) Observed that the `/opt/puppetlabs/server/apps/puppetserver/puppet-server-release.jar` file had been removed and that `/opt/puppetlabs/server/apps/puppetserver/puppetserver-release.jar` was now present.

I didn't test packages on other OSes that would use other types of init systems, but I'd hope the behavior would be the same anyway.  @haus, is that what you'd expect as well?

I honestly don't care all that much about the name of the uberjar file in the package.  I just had carried this change forward from @cprice404's original PR.  If we think there's too much risk associated with it, though, I'm open to reverting this part.